internal/notifier/google_helpers.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2025 The Flux authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package notifier
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+
+	"golang.org/x/oauth2"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/fluxcd/pkg/auth"
+	"github.com/fluxcd/pkg/auth/gcp"
+	"github.com/fluxcd/pkg/cache"
+
+	"github.com/fluxcd/notification-controller/api/v1beta3"
+)
+
+// GCPWorkloadIdentityConfig contains configuration for workload identity authentication.
+type GCPWorkloadIdentityConfig struct {
+	ServiceAccountName string
+	ProviderName       string
+	ProviderNamespace  string
+	ProxyURL           string
```

Comment:
Since we are introducing proxy support for authentication, we could also introduce proxy support for the whole operation, i.e. for interacting with the Pub/Sub API itself. For this we need to build a custom Google transport, like this:

https://github.com/fluxcd/source-controller/blob/c2b572bae03a53cef4355389494ea71097c6abfe/pkg/gcp/gcp.go#L90-L145